Backend/Models/OrderEntity/Order.cs

```diff
@@ -46,11 +49,46 @@ public class Order : ISoftDeletable
         [Required]
         public Warehouse Warehouse { get; set; }
 
+        [Required]
+        public int PharmacyId { get; set; }
+
+        [Required]
+        public Pharmacy Pharmacy { get; set; }
+
         [Required]
         public ICollection<OrderUser> OrderUsers { get; set; }
 
         [Required]
         [DefaultValue(false)]
         public bool IsSoftDeleted { get; set; } = false;
+
+        public Order() { }
+
+        public Order(CreateOrderDTO dto, string addressFrom, string addressTo)
+        {
+            AddressFrom = addressFrom;
+            AddressTo = addressTo;
+            CreationDate = dto.CreationDate;
+            DeliveryDate = dto.DeliveryDate;
+            OrderStateId = OrderStateId.Created;
+            Total = CalculateTotalAmount(dto);
+            WarehouseId = dto.WarehouseId;
+            PharmacyId = dto.PharmacyId;
+        }
+
+        public void UpdateFromDTO(CreateOrderDTO dto)
+        {
+            CreationDate = dto.CreationDate;
+            DeliveryDate = dto.DeliveryDate;
+            Total = CalculateTotalAmount(dto);
```

Comment:
Wouldn't this method override existing total amount? If we are updating the order (for example, adding new medicament to already existing ones), shouldn't we add the new total to already calculated one, instead of re-calculating from 0?